File: DTCore5.0-exp/DTCore/DataTools.Memory/Blob/BlobEnumerators.vb
'' ************************************************* ''
'' DataTools Visual Basic Utility Library 
''
'' Module: Enumerators for Blob
''         
'' Copyright (C) 2011-2020 Nathan Moschkin
'' All Rights Reserved
''
'' Licensed Under the Microsoft Public License   
'' ************************************************* ''

Namespace Memory


#Region "Enumerators"

    Public Class BlobEnumeratorByte
        Implements IEnumerator(Of Byte)

        Dim mm As Blob
        Dim pos As Integer = -1

        Friend Sub New(subj As Blob)
            mm = subj
        End Sub

        Public ReadOnly Property Current As Byte Implements IEnumerator(Of Byte).Current
            Get
                Current = mm.ByteAt(pos)
            End Get
        End Property

        Public ReadOnly Property Current1 As Object Implements IEnumerator.Current
            Get
                Current1 = mm.ByteAt(pos)
            End Get
        End Property

        Public Function MoveNext() As Boolean Implements IEnumerator.MoveNext
            pos += 1
            If pos >= mm.Length Then Return False Else Return True
        End Function

        Public Sub Reset() Implements IEnumerator.Reset
            pos = -1
        End Sub

#Region "IDisposable Support"
        Private disposedValue As Boolean ' To detect redundant calls

        ' IDisposable
        Protected Overridable Sub Dispose(disposing As Boolean)
            If Not Me.disposedValue Then
                mm = Nothing
            End If
            Me.disposedValue = True
        End Sub

        Protected Overrides Sub Finalize()
            ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(False)
            MyBase.Finalize()
        End Sub

        ' This code added by Visual Basic to correctly implement the disposable pattern.
        Public Sub Dispose() Implements IDisposable.Dispose
            ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
#End Region

    End Class

    Public Class BlobEnumeratorChar
        Implements IEnumerator(Of Char)

        Dim mm As Blob
        Dim pos As Integer = -1

        Friend Sub New(subj As Blob)
            mm = subj
        End Sub

        Public ReadOnly Property Current As Char Implements IEnumerator(Of Char).Current
            Get
                Current = mm.CharAt(pos)
            End Get
        End Property

        Public ReadOnly Property Current1 As Object Implements IEnumerator.Current
            Get
                Current1 = mm.CharAt(pos)
            End Get
        End Property

        Public Function MoveNext() As Boolean Implements IEnumerator.MoveNext
            pos += 1
            If pos >= CLng(mm.Length >> 1) Then Return False Else Return True
        End Function

        Public Sub Reset() Implements IEnumerator.Reset
            pos = -1
        End Sub

#Region "IDisposable Support"
        Private disposedValue As Boolean ' To detect redundant calls

        ' IDisposable
        Protected Overridable Sub Dispose(disposing As Boolean)
            If Not Me.disposedValue Then
                mm = Nothing
            End If
            Me.disposedValue = True
        End Sub

        Protected Overrides Sub Finalize()
            ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(False)
            MyBase.Finalize()
        End Sub

        ' This code added by Visual Basic to correctly implement the disposable pattern.
        Public Sub Dispose() Implements IDisposable.Dispose
            ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
#End Region

    End Class

#End Region '' Enumerators

End Namespace

File: DTCore5.0-exp/DTCore/DataTools.Memory/Blob/BlobEnumerators.cs
// ' ************************************************* ''
// ' DataTools Visual Basic Utility Library 
// '
// ' Module: Enumerators for Blob
// '         
// ' Copyright (C) 2011-2020 Nathan Moschkin
// ' All Rights Reserved
// '
// ' Licensed Under the Microsoft Public License   
// ' ************************************************* ''

using System;
using System.Collections.Generic;

namespace DataTools.Memory
{


    /* TODO ERROR: Skipped RegionDirectiveTrivia */
    public class BlobEnumeratorByte : IEnumerator<byte>
    {
        private Blob mm;
        private int pos = -1;

        internal BlobEnumeratorByte(Blob subj)
        {
            mm = subj;
        }

        public byte Current
        {
            get
            {
                byte CurrentRet = default;
                CurrentRet = mm.get_ByteAt(pos);
                return CurrentRet;
            }
        }

        public object Current
        {
            get
            {
                object CurrentRet = default;
                CurrentRet = mm.get_ByteAt(pos);
                return CurrentRet;
            }
        }

        public bool MoveNext()
        {
            pos += 1;
            if (pos >= mm.Length)
                return false;
            else
                return true;
        }

        public void Reset()
        {
            pos = -1;
        }

        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private bool disposedValue; // To detect redundant calls

        // IDisposable
        protected virtual void Dispose(bool disposing)
        {
            if (!disposedValue)
            {
                mm = null;
            }

            disposedValue = true;
        }

        ~BlobEnumeratorByte()
        {
            // Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(false);
        }

        // This code added by Visual Basic to correctly implement the disposable pattern.
        public void Dispose()
        {
            // Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(true);
            GC.SuppressFinalize(this);
        }
        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }

    public class BlobEnumeratorChar : IEnumerator<char>
    {
        private Blob mm;
        private int pos = -1;

        internal BlobEnumeratorChar(Blob subj)
        {
            mm = subj;
        }

        public char Current
        {
            get
            {
                char CurrentRet = default;
                CurrentRet = mm.get_CharAt(pos);
                return CurrentRet;
            }
        }

        public object Current
        {
            get
            {
                object CurrentRet = default;
                CurrentRet = mm.get_CharAt(pos);
                return CurrentRet;
            }
        }

        public bool MoveNext()
        {
            pos += 1;
            if (pos >= mm.Length >> 1)
                return false;
            else
                return true;
        }

        public void Reset()
        {
            pos = -1;
        }

        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private bool disposedValue; // To detect redundant calls

        // IDisposable
        protected virtual void Dispose(bool disposing)
        {
            if (!disposedValue)
            {
                mm = null;
            }

            disposedValue = true;
        }

        ~BlobEnumeratorChar()
        {
            // Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(false);
        }

        // This code added by Visual Basic to correctly implement the disposable pattern.
        public void Dispose()
        {
            // Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(true);
            GC.SuppressFinalize(this);
        }
        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }

    /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
}